cpp/ql/lib/semmle/code/cpp/commons/Buffer.qll

```diff
@@ -10,11 +10,22 @@ import semmle.code.cpp.dataflow.DataFlow
  *   char data[1]; // v
  * };
  * ```
- * This requires that `v` is an array of size 0 or 1.
+ * or
+ * ```
+ * struct myStruct { // c
+ *   int amount;
+ *   char data[]; // v
+ * };
+ * ```
+ * This requires that `v` is an array of size 0 or 1, or that the array has no size.
  */
 predicate memberMayBeVarSize(Class c, MemberVariable v) {
   c = v.getDeclaringType() and
-  v.getUnspecifiedType().(ArrayType).getArraySize() <= 1
+  exists(ArrayType t | t = v.getUnspecifiedType() |
+    t.getArraySize() <= 1
+    or
+    not exists(t.getArraySize())
```

Comment:
I think these two lines could be simplified to:
```
not t.getArraySize() > 1
```

Reply:
Indeed, thanks for this. I'll apply this change.

Reply:
Done.